### PR TITLE
[graphql/rpc] owner interface and stubs

### PR DESCRIPTION
**Pull request by oxade**
## Description 

Adds the `Owner` interface and the required stubs to get it to work

## Test Plan 

Snapshot

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by wlmyng — APPROVED** [review]

File: crates/sui-graphql-rpc/src/types/stake.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    async fn id(&self) -> ID {
+        unimplemented!()
+    }
+}
+
+#[allow(unreachable_code)]
+#[allow(unused_variables)]
+#[Object]
+impl StakeConnection {
+    async fn id(&self) -> ID {
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/stake.rs:22]: Do the connections themselves need `id`?

File: crates/sui-graphql-rpc/src/types/name_service.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+use async_graphql::*;
+
+pub(crate) struct NameService;
+pub(crate) struct NameServiceConnection;
+
+#[allow(unreachable_code)]
+#[allow(unused_variables)]
+#[Object]
+impl NameService {
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/name_service.rs:12]: In the schema `NameService` is just a `String`, i.e. it has no `id`, is this change intentional?

File: crates/sui-graphql-rpc/src/types/balance.rs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    async fn id(&self) -> ID {
+        unimplemented!()
+    }
+}
+
+#[allow(unreachable_code)]
+#[allow(unused_variables)]
+#[Object]
+impl BalanceConnection {
+    async fn id(&self) -> ID {
```
[the remark this answers is not in the document]
Comment (amnn) [reply crates/sui-graphql-rpc/src/types/balance.rs:22]: It's different -- it's an opaque/generic ID used to refer to any entity that the GraphQL server provides.  Eventually there will be a `node` query on the server that takes an `ID` and provides the content of that node (whether it's an `Object` or a `TransactionBlock`, or a `Balance`, etc).

File: crates/sui-graphql-rpc/src/types/owner.rs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+use crate::types::name_service::*;
+use crate::types::object::*;
+use crate::types::stake::*;
+use crate::types::sui_address::SuiAddress;
+use async_graphql::*;
+
+use super::address::Address;
+
+#[derive(Interface)]
+#[graphql(
```
[the remark this answers is not in the document]
Comment (amnn) [reply crates/sui-graphql-rpc/src/types/owner.rs:15]: I believe it's needed because `Owner` is an interface (in the GraphQL schema).

File: crates/sui-graphql-rpc/src/types/stake.rs (hunk 1 above)
Comment (oxade) [reply crates/sui-graphql-rpc/src/types/stake.rs:22]: indeed this is not needed. But the graphql lib will complain if a type is defined without any fields/methods. I initially put these as a placeholder but should remove them to avoid confusion.

File: crates/sui-graphql-rpc/src/types/name_service.rs (hunk 2 above)
Comment (oxade) [reply crates/sui-graphql-rpc/src/types/name_service.rs:12]: see comment on placeholder to appease graphql macro

Comment (oxade) [reply crates/sui-graphql-rpc/src/types/name_service.rs:12]: true, will remove
